src/Soil/energy_hydrology.jl

```diff
@@ -397,3 +417,205 @@ function ClimaLSM.source!(
+        p,
+    ) where {FT}
+
+Returns the surface emissivity field of the 
+`EnergyHydrology` soil model.
+"""
+function ClimaLSM.surface_emissivity(
+    model::EnergyHydrology{FT},
+    Y,
+    p,
+) where {FT}
+    return model.parameters.emissivity
+end
+
+"""
+    ClimaLSM.surface_albedo(
+        model::EnergyHydrology{FT},
+        Y,
+        p,
+    ) where {FT}
+
+Returns the surface albedo field of the 
+`EnergyHydrology` soil model.
+"""
+function ClimaLSM.surface_albedo(model::EnergyHydrology{FT}, Y, p) where {FT}
+    return model.parameters.albedo
+end
+
+"""
+    ClimaLSM.surface_air_density(
+        atmos::PrescribedAtmosphere{FT},
+        model::EnergyHydrology{FT},
+        Y,
+        p,
+        t,
+        T_sfc
+    ) where {FT}
+
+Returns the surface air density field of the 
+`EnergyHydrology` soil model for the 
+`PrescribedAtmosphere` case.
+
+This assumes the ideal gas law and hydrostatic
+balance to estimate the air density at the surface
+from the values of surface temperature and the atmospheric
+thermodynamic state, 
+because the surface air density is not a prognostic
+variable of the soil model.
+"""
+function ClimaLSM.surface_air_density(
+    atmos::PrescribedAtmosphere{FT},
+    model::EnergyHydrology{FT},
+    Y,
+    p,
+    t,
+    T_sfc,
+) where {FT}
+
+    thermo_params =
+        LSMP.thermodynamic_parameters(model.parameters.earth_param_set)
+    ts_in = construct_atmos_ts(atmos, t, thermo_params)
+    return compute_ρ_sfc.(Ref(thermo_params), Ref(ts_in), T_sfc)
+end
+
+"""
+    ClimaLSM.surface_specific_humidity(
+        model::EnergyHydrology{FT},
+        Y,
+        p,
+        T_sfc,
+        ρ_sfc
+    ) where {FT}
+
+Returns the surface specific humidity field of the 
+`EnergyHydrology` soil model.
+
+This models the surface specific humidity as
+the saturated value multiplied by
+the factor `exp(ψ_sfc g M_w/(RT_sfc))` in accordance
+with the Clausius-Clapeyron equation,
+where `ψ_sfc` is the matric potential at the surface,
+`T_sfc` the surface temperature, `g` the gravitational
+acceleration on the surface of the Earth, `M_w` the molar
+mass of water, and `R` the universal gas constant.
+"""
+function ClimaLSM.surface_specific_humidity(
+    model::EnergyHydrology{FT},
+    Y,
+    p,
+    T_sfc,
+    ρ_sfc,
+) where {FT}
+    g = LSMP.grav(model.parameters.earth_param_set)
+    R = LSMP.gas_constant(model.parameters.earth_param_set)
+    M_w = LSMP.molar_mass_water(model.parameters.earth_param_set)
+    thermo_params =
+        LSMP.thermodynamic_parameters(model.parameters.earth_param_set)
+    face_space = ClimaLSM.Domains.obtain_face_space(model.domain.space)
+    N = ClimaCore.Spaces.nlevels(face_space)
+    interp_c2f = ClimaCore.Operators.InterpolateC2F(
+        top = ClimaCore.Operators.Extrapolate(),
+        bottom = ClimaCore.Operators.Extrapolate(),
+    )
+    surface_space = ClimaLSM.Domains.obtain_surface_space(model.domain.space)
+    ψ_sfc = ClimaCore.Fields.Field(
```

Comment:
we do this three times. make a function for it?

Reply:
we also do it in the bucket and soil+surface water models. So, perhaps do in a follow PR. I will open an issue